platform/darwin/src/gl.cpp

```diff
@@ -0,0 +1,187 @@
+#include <mbgl/platform/gl.hpp>
+
+#include "TargetConditionals.h"
+
+#if TARGET_OS_IPHONE
+    #include <OpenGLES/ES2/gl.h>
+    #include <OpenGLES/ES2/glext.h>
+#elif TARGET_OS_SIMULATOR
+    #include <OpenGLES/ES2/gl.h>
+    #include <OpenGLES/ES2/glext.h>
+#elif TARGET_OS_MAC
+    #include <OpenGL/OpenGL.h>
+    #include <OpenGL/gl.h>
+    #include <OpenGL/glext.h>
+#else
+    #error Unsupported Apple platform
+#endif
+
+namespace mbgl {
+namespace platform {
+
+gl::GLFunctionPointers getGLFunctionPointers() {
```

Comment:
This seems like the worst of both worlds: We still need to link to OpenGL.dylib, but are also manually copying around the function pointers and calling through them instead of directly linking.

Reply:
Agree, let's make it a linking time thingy instead of copying pointers.